client/CPlayerInterface.cpp

```diff
@@ -2637,9 +2637,39 @@ void CPlayerInterface::doMoveHero(const CGHeroInstance* h, CGPath path)
 		ETerrainType newTerrain;
 		int sh = -1;
 
-		const TerrainTile * curTile = cb->getTile(CGHeroInstance::convertPosition(h->pos, false));
+		auto allowedToStopAt = [this,h]{ //make sure hero won't able to stop on blocked tile while flying
+			bool canStopAt = true;
+			const TerrainTile * curTile = cb->getTile(CGHeroInstance::convertPosition(h->pos, false));

```

Comment:
> Player interface should never allow players to stop in mid-air even if they insist to do so.
Actually in H3 player allowed to spot fly movement if tile under him is visitable and not blocked.
Why shouldn't we allow that too? Or it's not what you mean?